types/decimal.go

```diff
@@ -339,7 +339,9 @@ func (d Dec) QuoInt64(i int64) Dec {
 // using Newton's method (where n is positive). The algorithm starts with some guess and
 // computes the sequence of improved guesses until an answer converges to an
 // approximate answer.  It returns `|d|.ApproxRoot() * -1` if input is negative.
-func (d Dec) ApproxRoot(root uint64) (guess Dec, err error) {
+// The maximum number of iterations is required as a backup boundary condition for
+// cases where the answer never converges enough to satisfy the main condition.
+func (d Dec) ApproxRoot(root, maxIters uint64) (guess Dec, err error) {
```

Comment:
I personally don't feel comfortable with this code. If this is necessary by the application, why not just define it there? We don't even use this in the SDK or relevant applications (e.g. gaia and simapp).

Reply:
I understand. My reasoning is that it's nice to have root calculation functionality built into the SDK alongside the Dec type, but as you said one could implement it on the app side of things if they need it.

I'm curious if it's the solution approach that you don't feel comfortable with or the function/s in general?

Reply:
If we want to remove it outright, I can get that done instead of the above solution.

Reply:
I think having a root function is pretty important for a Decimal/Number library, as its a pretty common operation.  And implementing a solid version in the library itself would be better than having different applications all having their own potentially broken implementation (oops 😅).